src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -745,20 +801,42 @@ await ExecuteWithFailOverPolicyAsync(
 
                 Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
                 SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
+
+                var selectedKeyValueWatchers = new List<KeyValueWatcher>();
+
+                if (_options.RefreshOptions.RegisterAllEnabled)
+                {
+                    foreach (KeyValueIdentifier keyValueIdentifier in watchedSelectedKeyValueCollections.Keys)
+                    {
+                        selectedKeyValueWatchers.Add(new KeyValueWatcher()
+                        {
+                            Key = keyValueIdentifier.Key,
+                            Label = keyValueIdentifier.Label,
+                            RefreshInterval = _options.RefreshOptions.RefreshInterval
+                        });
+                    }
+                }
+
+                // Update the next refresh time for all refresh registered settings and feature flags
+                foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers.Concat(_options.FeatureFlagWatchers).Concat(selectedKeyValueWatchers))
+                {
+                    UpdateNextRefreshTime(changeWatcher);
+                }
+
                 _watchedSettings = watchedSettings;
+                _watchedSelectedKvCollections = watchedSelectedKeyValueCollections;
+                _watchedFeatureFlagCollections = watchedFeatureFlagCollections;
                 _mappedData = mappedData;
+                _selectedKvWatchers = selectedKeyValueWatchers;
             }
         }
 
-        private async Task<Dictionary<string, ConfigurationSetting>> LoadSelectedKeyValues(ConfigurationClient client, CancellationToken cancellationToken)
+        private async Task<Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>>> LoadSelected(Dictionary<string, ConfigurationSetting> existingData, IEnumerable<KeyValueSelector> selectors, ConfigurationClient client, CancellationToken cancellationToken)
         {
-            var serverData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
-
-            // Use default query if there are no key-values specified for use other than the feature flags
-            bool useDefaultQuery = !_options.KeyValueSelectors.Any(selector => selector.KeyFilter == null ||
-                !selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker));
+            var watchedCollections = new Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>>();
 
-            if (useDefaultQuery)
+            // Use default query if there are no key-values specified for use
+            if (!_options.KeyValueSelectors.Any())
```

Comment:
Since `LoadSelected` is called with kvs and feature flags, will we load from app config service 2 times when there are no key value selectors specified?

Reply:
Fixed now